Remove setting of PT_HPU_LAZY_MODE=2 in training_args.py

vivekgoe opened the pull request:
# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->
Cleanup code in training_args.py to remove setting of PT_HPU_LAZY_MODE=2 (this is not supported SynapseAI release 1.13.0 onwards).

<!-- Remove if not applicable -->

Fixes # (issue)


## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

regisss commented on optimum/habana/transformers/training_args.py:
```diff
@@ -648,12 +648,11 @@ def _setup_devices(self) -> "torch.device":
                 if os.getenv("PT_HPU_LAZY_MODE", "1") != "0":
-                    os.environ["PT_HPU_LAZY_MODE"] = "2"
-                logger.info("Enabled eager mode because use_lazy_mode=False.")
+                    assert (
+                        False
+                    ), "Lazy mode or compile mode not enabled => eager mode should be enabled using PT_HPU_LAZY_MODE=0"
```
```suggestion
                    raise ValueError(
                        "Lazy mode or compile mode not enabled => eager mode should be enabled using PT_HPU_LAZY_MODE=0")
```
Let's rather raise an issue, I feel like it makes the error message clearer.

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/optimum-habana/pr_625). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

vivekgoe commented:
@regisss Since I pushed this PR from a fork which has a different name "HabanaAI/optimum-habana-fork" instead of usual "<user-name>/optimum-habana , there was a failure in Build PR documentation step. I added a small change to Dockerfile to fix this.
Also with this change only eager mode test we have (test_trainer.py/test_eager_mode) started failing (throwing new assert we have added in this PR). This can be fixed only when whole test suite is switched to PT_HPU_LAZY_MODE=0, reason is that this environment variable can't be updated once Synapse SW is loaded. For now I have added a skip for this test, please suggest if can think of a better solution.

vivekgoe commented on tests/test_trainer.py:
```diff
@@ -706,6 +707,7 @@ def setUp(self):
         self.n_epochs = args.num_train_epochs
         self.batch_size = args.train_batch_size
 
+    @mark.skip("Skip this test until PT_HPU_LAZY_MODE=0 is set as default for all tests")
```
In near future (when we deprecate Lazy mode) it will be set as default in SynapseAI. 

Until then only way to run this test is if we can somehow set this environment variable before any import calls to habana_frameworks, because once habana_framworks is loaded it sets default PT_HPU_LAZY_MODE value (=1) and then changing environment variable with a different value has no impact.  

Since this test runs as a part of full test-suite, 1st test run will load habana_framworks and I do not see a good way of changing environment variable value later.

regisss reviewed (approved):
LGTM
